javaslang/src/main/java/javaslang/collection/Stream.java

```diff
@@ -1414,28 +1408,19 @@ private Object readResolve() {
     }
 
     /**
-     * Non-empty {@code Stream}, consisting of a {@code head}, a {@code tail} and an optional
-     * {@link java.lang.AutoCloseable}.
+     * Non-empty {@code Stream}, consisting of a {@code head}, and {@code tail}.
      *
      * @param <T> Component type of the Stream.
      * @since 1.1.0
      */
-    // DEV NOTE: class declared final because of serialization proxy pattern.
-    // (see Effective Java, 2nd ed., p. 315)
-    final class Cons<T> implements Stream<T>, Serializable {
+    abstract class Cons<T> implements Stream<T> {
```

Comment:
@danieldietrich I know you don't like abstract classes in such context, but but here it simplifies the code